File: Python/Dijkstra.py
import timeit

class Graph():

    def __init__(self, vertices):
        self.V = vertices
        self.graph = [[0 for column in range(vertices)]
                      for row in range(vertices)]

    def printSolution(self, dist):
        print ("Vertex \tDistance from Source")
        for node in range(self.V):
            print ("  ", node, "\t\t\t", dist[node])

    def minDistance(self, dist, sptSet):

        min = 99999999

        for v in range(self.V):
            if dist[v] < min and sptSet[v] == False:
                min = dist[v]
                min_index = v

        return min_index

    def dijkstra(self, src):

        dist = [99999999] * self.V
        dist[src] = 0
        sptSet = [False] * self.V

        for _ in range(self.V):

            u = self.minDistance(dist, sptSet)

            sptSet[u] = True

            for v in range(self.V):
                if self.graph[u][v] > 0 and sptSet[v] == False and \
                        dist[v] > dist[u] + self.graph[u][v]:
                    dist[v] = dist[u] + self.graph[u][v]

        self.printSolution(dist)


n = int(input("Enter the no of vertex: "))
g = Graph(n)
print(f"Enter the adjacency matrix of {n}x{n}: ")
g.graph = [list(map(int, input().split())) for _ in range(n)]

s = timeit.default_timer()
g.dijkstra(0);
e = timeit.default_timer()

print("\nExecution time is {} seconds".format(e-s))

File: JAVA/Dijkstra.java
import java.util.*;

class dijkstra {

	int minDistance(int dist[], Boolean sptSet[], int V) {
		int min = Integer.MAX_VALUE, min_index = -1;

		for (int v = 0; v < V; v++)
			if (sptSet[v] == false && dist[v] <= min) {
				min = dist[v];
				min_index = v;
			}

		return min_index;
	}

	void printSolution(int dist[], int V) {
		System.out.println("Vertex \t\t Distance from Source");
		for (int i = 0; i < V; i++)
			System.out.println(i + " \t\t " + dist[i]);
	}

	void dijkstraFun(int graph[][], int src, int V) {
		int dist[] = new int[V];

		Boolean sptSet[] = new Boolean[V];

		for (int i = 0; i < V; i++) {
			dist[i] = Integer.MAX_VALUE;
			sptSet[i] = false;
		}

		dist[src] = 0;

		for (int count = 0; count < V - 1; count++) {
			int u = minDistance(dist, sptSet, V);

			sptSet[u] = true;

			for (int v = 0; v < V; v++)

				if (!sptSet[v] && graph[u][v] != 0 && dist[u] != Integer.MAX_VALUE && dist[u] + graph[u][v] < dist[v])
					dist[v] = dist[u] + graph[u][v];
		}

		printSolution(dist, V);
	}

	public static void main(String[] args) {
		Scanner scan = new Scanner(System.in);

		System.out.print("Enter the no of vertex: ");
		int n = scan.nextInt();
		System.out.println();

		System.out.print("Enter the matrix: ");

		int graph[][] = new int[n][n];

		for (int i = 0; i < n; ++i) {
			for (int j = 0; j < n; ++j) {
				graph[i][j] = scan.nextInt();
			}
		}
		scan.close();
		dijkstra t = new dijkstra();

		long startTime = System.nanoTime();
		t.dijkstraFun(graph, 0, n);
		long elapsedTime = System.nanoTime() - startTime;

		double seconds = (double) elapsedTime / 1_000_000_000.0;
		System.out.println("Execution Time : " + seconds);
	}
}